lang/src/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -292,6 +299,29 @@ private class DecimalExprValue(val ion: IonSystem, val value: BigDecimal): Scala
     override fun ionValueFun(): IonValue = ion.newDecimal(value)
 }
 
+/**
+ * [ExprValue] to represent DATE in PartiQL.
+ * [LocalDate] represents date without time and time zone.
+ */
+private class DateExprValue(val ion: IonSystem, val value: LocalDate): ScalarExprValue() {
+
+    private val PARTIQL_DATE_ANNOTATION = "partiql_date"
+
+    private fun createIonDateStruct() =
+        ion.newEmptyStruct().apply {
+            add("year", ion.newInt(value.year))
+            add("month", ion.newInt(value.monthValue))
+            add("day", ion.newInt(value.dayOfMonth))
+        }.apply {
+            addTypeAnnotation(PARTIQL_DATE_ANNOTATION)
+        }.seal()
```

Comment:
Let's make this `$partiql_date` as the annotation.

Also, how do we feel about using a timestamp with an annotation for this?  Does this limit us in any way?

Reply:
I guess we could do that. Although I do have one minor doubt here. What if someone uses the `date`'s `ionvalue` as `timestamp` with `day's precision`.  Is there any potential issue with that?
For now, I have changed it to use timestamp.